### DISCUSSION TITLE
Mark commit as base commit for rebase

### DISCUSSION THREAD

**Discussion by mloskot** [jesseduffield/lazygit#3154] [Q&A] [answered]
Hi, there is this `B` command to "Mark commit as base commit for rebase" but I'm failing to understand the uses of it.
Could anyone present a use case when this command can be useful?

**Comment by stefanhaller** [comment, marked as answer]
Just one example out of many: suppose you have two release branches, "main" and "develop". You start a feature branch off of "develop", and then after a while you decide you want to move it to "main" instead. You can't just select main and hit "r", because this would not only rebase the commits of your feature branch, but also all commits between it and main, so the result would not be what you want. In this situation you would select the commit on "develop" that your feature branch forks off of, mark it as the base commit for rebase, and then select main and hit "r".

Basically you need it any time you would have used "rebase --onto" on the command line. If you google for that, you'll find many other examples.

**Reply by mloskot** [reply]
@stefanhaller Thank you for the excellent explanation. The key part that prevented me from connecting the right dots was the reference between that `B` shortcut and the `rebase --onto`.
